File: HSharp_v0.4_Package/benchmarks/bench_string.py
# Python 3 Benchmark 3: String Concatenation
# Build a long string by concatenation
# Equivalent to bench_string.hto

import time

n = 50000
s = ""
i = 0
t0 = time.perf_counter()
while i < n:
    s = s + "abcdef"
    i = i + 1
elapsed = time.perf_counter() - t0
print("len=" + str(len(s)))
print("time_ms=" + str(round(elapsed * 1000, 3)))

File: HSharp_v0.4_Package/benchmarks/bench_string.cpp
// C++ Benchmark 3: String Concatenation
// Build a long string by concatenation
// Equivalent to bench_string.hto
// Build: g++ -O2 -o bench_string bench_string.cpp

#include <cstdio>
#include <chrono>
#include <string>

int main() {
    const int n = 50000;
    std::string s;
    auto t0 = std::chrono::high_resolution_clock::now();
    for (int i = 0; i < n; i = i + 1) {
        s = s + "abcdef";
    }
    auto t1 = std::chrono::high_resolution_clock::now();
    double ms = std::chrono::duration<double, std::milli>(t1 - t0).count();
    printf("len=%zu\n", s.size());
    printf("time_ms=%.3f\n", ms);
    return 0;
}
